### ISSUE TITLE
Hard negative mining

### ISSUE THREAD

**Issue by tuanphan09** [Walleclipse/Deep_Speaker-speaker_recognition_system#47]
Based on your report, training with random batch was better than using hard negative mining, right?

**Comment by Walleclipse** [comment]
Actually not.
The better strategy is, in the earlier stage of the learning  (0\~80k steps) using softmax pretraining or random batch learning, then (80\~110k learning steps) using hard negative mining.  In [report](https://github.com/Walleclipse/Deep_Speaker-speaker_recognition_system/blob/master/deep_speaker_report.pdf). EER for random batch strategy is 8%, EER for hard negative mining is 6%.

**Comment by tuanphan09** [comment]
Thank you.

Could you tell me how did you get the 17k pretrained model in checkpoint model (with/without hard negative mining, how many steps?)? I'm trying to compare my model with yours and working on this task with GAN like [this paper](https://arxiv.org/abs/1803.09059)

**Comment by Walleclipse** [comment]
You can get pretrained model in the following steps.
1. In `constant.py` , set `PRE_TRAIN = False`
2. run `pretraining.py`
I approximately run 80,000 steps in softmax pretraining.

I am not familiar with GAN, and I look forward to seeing your results on GAN.
